src/Assert.php

```diff
@@ -712,6 +718,8 @@ public static function ip($value, $message = '')
     }
 
     /**
+     * @psalm-assert string $value
```

Comment:
`filter_var` also allows objects that can be cast to string, so this isn't correct

Reply:
NNNNNNNGGGGGHHHHH PHP!!!! >.<

Ok, will have to revert this one then :-(

Reply:
🙃

Reply:
This won't work yet, due to vimeo/psalm#2452

Will add to the patch for now, knowing that it will fail until upstream issue has some sort of resolution (which may also be to remove the type assertion here)